src/main/java/org/skriptlang/skript/misc/colours/ColourUtils.java

```diff
@@ -0,0 +1,344 @@
+		amount = Math.max(1, Math.min(100, amount));
+		float[] hsl = rgbToHsl(color);
+		hsl[2] += (1f - hsl[2]) * (amount / 100f);
+		hsl[2] = Math.min(1f, hsl[2]);
+		return hslToRgb(hsl);
+	}
+
+	/**
+	 * Rotates the hue of a {@link Color} by a given degree.
+	 *
+	 * @param color the {@link Color} to rotate the hue of
+	 * @param degrees the number of degrees to rotate the hue by
+	 * @return the hue-rotated colour
+	 */
+	public static Color rotateHue(Color color, int degrees) {
+		float[] hsl = rgbToHsl(color);
+		hsl[0] = (hsl[0] + degrees / 360f) % 1f;
+		if (hsl[0] < 0f)
+			hsl[0] += 1f;
+		return hslToRgb(hsl);
+	}
+
+	/**
+	 * Adjusts the brightness of a {@link Color} by an amount from -100 to 100.
+	 * This is similar to shading and tinting, but is slightly different.
+	 *
+	 * @param color the {@link Color} to adjust the brightness of
+	 * @param amount the amount to adjust the brightness by (-100 - 100)
+	 * @return the brightness-adjusted colour
+	 */
+	public static ColorRGB adjustBrightness(Color color, int amount) {
+		amount = Math.max(-100, Math.min(100, amount));
+		float[] hsb = rgbToHsb(color);
+		float factor = amount / 100f;
+		hsb[2] = hsb[2] + hsb[2] * factor;
+		hsb[2] = Math.max(0f, Math.min(1f, hsb[2]));
+		return hsbToRgb(hsb);
+	}
+
+	/**
+	 * Converts a {@link Color} to HSB (hue, saturation, brightness).
+	 * This is different to {@link #rgbToHsl(Color)}.
+	 *
+	 * @param color the {@link Color} to convert
+	 * @return a float array representing the HSB values
+	 */
+	private static float[] rgbToHsb(Color color) {
+		float r = color.getRed() / 255f;
+		float g = color.getGreen() / 255f;
+		float b = color.getBlue() / 255f;
+		float max = Math.max(r, Math.max(g, b));
+		float min = Math.min(r, Math.min(g, b));
+		float delta = max - min;
+		float h = 0f;
+		float s = max == 0f ? 0f : delta / max;
+		float v = max;
+		if (delta != 0f) {
+			if (max == r) {
+				h = ((g - b) / delta) % 6f;
+			} else if (max == g) {
+				h = ((b - r) / delta) + 2f;
+			} else {
+				h = ((r - g) / delta) + 4f;
+			}
+			h *= 60f;
+			if (h < 0f) h += 360f;
+		}
+		h /= 360f;
+		return new float[]{ h, s, v };
+	}
+
+	/**
+	 * Converts HSB values to a {@link ColorRGB} object.
+	 * This is different to {@link #hslToRgb(float[])}.
+	 *
+	 * @param hsb a float array representing HSB values
+	 * @return a {@link ColorRGB} object given the HSB values
+	 */
+	private static ColorRGB hsbToRgb(float[] hsb) {
+		float h = hsb[0], s = hsb[1], v = hsb[2];
+		float r = 0f, g = 0f, b = 0f;
+		int i = (int) (h * 6f);
+		float f = h * 6f - i;
+		float p = v * (1f - s);
+		float q = v * (1f - f * s);
+		float t = v * (1f - (1f - f) * s);
+		switch (i % 6) {
+			case 0:
+				r = v; g = t; b = p; break;
+			case 1:
+				r = q; g = v; b = p; break;
+			case 2:
+				r = p; g = v; b = t; break;
+			case 3:
+				r = p; g = q; b = v; break;
+			case 4:
+				r = t; g = p; b = v; break;
+			case 5:
+				r = v; g = p; b = q; break;
```

Comment:
we can afford more than one character per variable name
this applies to all the other methods too